src/Arcus.BackgroundJobs.DataBricks/DatabricksJobMetricsJobSchedulerOptions.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System;
+using System.Threading.Tasks;
+using Arcus.Security.Core;
+using CronScheduler.Extensions.Scheduler;
+using GuardNet;
+using Microsoft.Azure.Databricks.Client;
+
+namespace Arcus.BackgroundJobs.Databricks
+{
+    /// <summary>
+    /// Options to configure how the <see cref="DatabricksJobMetricsJob"/> scheduled job.
+    /// </summary>
+    public class DatabricksJobMetricsJobSchedulerOptions : SchedulerOptions
+    {
+        private string _baseUrl, _tokenSecretKey;
+
+        /// <summary>
+        /// Gets or sets the URL of the Databricks location.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="value"/> is blank.</exception>
+        public string BaseUrl
+        {
+            get => _baseUrl;
+            set
+            {
+                Guard.NotNullOrWhitespace(value, nameof(value));
+                _baseUrl = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the secret key to retrieve the token from the registered <see cref="ISecretProvider"/>.
+        /// </summary>
+        /// <exception cref="ArgumentException">Thrown when <paramref name="value"/> is blank.</exception>
+        public string TokenSecretKey
+        {
+            get => _tokenSecretKey;
+            set
+            {
+                Guard.NotNullOrWhitespace(value, nameof(value));
+                _tokenSecretKey = value;
+            }
+        }
+
+        /// <summary>
+        /// Gets the additional user options which configures the <see cref="DatabricksJobMetricsJob"/> scheduled job.
+        /// </summary>
+        public DatabricksJobMetricsJobOptions UserOptions { get; private set; } = new DatabricksJobMetricsJobOptions();
+
+        /// <summary>
+        /// Sets the additional user options in a <see cref="SchedulerOptions"/> context.
+        /// </summary>
+        /// <param name="options">The additional user-options to set.</param>
+        internal void SetUserOptions(DatabricksJobMetricsJobOptions options)
+        {
+            Guard.NotNull(options, nameof(options));
+
+            UserOptions = options;
+            CronSchedule = options.IntervalInMinutes == 1 ? "@every_minute" : $"*/{options.IntervalInMinutes} * * * *";
+        }
+
+        /// <summary>
+        /// Creates an <see cref="DatabricksClient"/> instance using the predefined values.
+        /// </summary>
+        /// <param name="secretProvider">The provider to retrieve the token during the creation of the instance.</param>
+        internal async Task<DatabricksClient> CreateDatabricksClientAsync(ISecretProvider secretProvider)
```

Comment:
Suggestion - protected virtual so that it can be extended? For example it could be used to unit test the job and inject mocked client with runs to verify if ILogger was called

Reply:
Hmmm, the scenario that you talk about _"inject runs in client to verify if logger was called"_ is actually the scenario of the integration test. The added-value here would be that the unit test interact with the job directly (ex. calling `.ExecuteAsync` directly) so that we can _always_ run it and not skip it like the integration test is now.
Ok, I see the added-value for this! I'll update this and add a unit test.

Reply:
Nice catch btw!

Reply:
Hmm, that doesn't seem as easy as I thought. Since we're interacting with the job itself, we don't have control over the time windows, therefore, we should 'simulate' a delay to increase the time between runs, which would lead maybe to too much setup for something that's already in the integration suite.

Reply:
Or we could inject an interface of the `IDatabricksInfoProvider` so we can skip the time calculation.

Reply:
We have integration tests to effectively test the integration but I think the majority of this can already be tested with unit tests.

Well you could argue that the windows should be a property on the job itself as well which feels more natural. If that's the case, you could also tweak them then to manipulate your test, but that's not the reason why I'm suggesting it.

Reason why I'd like to have it as a unit test is, well frankly it's fairly easy to do and we mock out the Databricks part. The integration test is solely to test that part which we cannot do with a unit test.

Reply:
> Or we could inject an interface of the `IDatabricksInfoProvider` so we can skip the time calculation.

No because you have to create it as part of your options.

Reply:
Had to made it `public virtual` for test.